agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -100,6 +101,35 @@ namespace beerocks {
 #define FSM_IS_IN_STATE(eState) (m_eFSMState == EState::eState)
 #define FSM_CURR_STATE_STR s_arrStates[int(m_eFSMState)]
 
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Helper Functions /////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+template <class SlavesContainer, class MAC>
+std::shared_ptr<backhaul_manager::sRadioInfo> findRadioInfo(const SlavesContainer &slave_container,
+                                                            const MAC &mac)
+{
+    // for each radio info in the container
+    //      for each associated client
+    //          find if there is an entry with the given mac within this associated client
+
+    for (const auto &slave : slave_container) {
+        if (slave) {
+            LOG(DEBUG) << "looking into radio info for (hostap_iface): " << slave->hostap_iface;
+
+            for (const auto &client : slave->associated_clients_map) {
+                LOG(DEBUG) << "client: " << client.first;
+
+                if (client.second.find(mac) != client.second.end()) {
+                    return slave;
+                }
+            }
+        }
```

Comment:
Aligned to our coding standards:

```suggestion
        if (!slave) {
            continue;
        }
        
        LOG(DEBUG) << "looking into radio info for (hostap_iface): " << slave->hostap_iface;

        for (const auto &client : slave->associated_clients_map) {
            LOG(DEBUG) << "client: " << client.first;

            if (client.second.find(mac) != client.second.end()) {
                return slave;
            }
        }
```